docs/BuildExtensibility.md

```diff
@@ -0,0 +1,141 @@
+![UI5 icon](https://raw.githubusercontent.com/SAP/ui5-tooling/master/docs/images/UI5_logo_wide.png)
+
+# UI5 Build Extensibility
+
+The UI5 Tooling enables it to define a custom build for your UI5 project by adding additional tasks to the  [standard task list](https://github.com/SAP/ui5-builder/blob/master/README.md#tasks).
+
+## Configuration
+
+The additional build tasks can be defined in the project configuration within the `ui5.yaml` file.
+
+### Example: Configuration of project 'my.library'
+
+````yaml
+---
+# In this example configuration two custom tasks are defined: 'babel' and 'generateMarkdownFiles'.
+specVersion: "0.1"
+type: library
+metadata:
+    name: my.library
+builder:
+    customTasks:
+    - name: babel
+      beforeTask: generateComponentPreload
+    - name: generateMarkdownFiles
+      afterTask: uglify
+      configuration:
+        color: blue
+````
+
+In the above example, when building the library `my.library` the `babel` task will be executed before the standard task `generateComponentPreload`. Another custom task called `generateMarkdownFiles` is then executed immediatly after the standard task `uglify`.
+
+### Example: Connect multiple custom tasks
+
+You can also connect multiple custom task with each other. If you do, please be aware that the order of your definitions is important. You have to make sure that the task is defined before you set it as `beforeTask` or `afterTask`.
+
+````yaml
+---
+# In this example 'myCustomTask2' gets executed after 'myCustomTask1'.
+specVersion: "0.1"
+type: library
+metadata:
+    name: my.library
+builder:
+    customTasks:
+    - name: myCustomTask1
+      beforeTask: generateComponentPreload
+    - name: myCustomTask2
+      afterTask: myCustomTask1
+````
+
+## Custom Task Extension
+
+A custom task extension consists of a `ui5.yaml` and a task implementation.
```

Comment:
Maybe add second sentence: `It can be a standalone module or part of an existing UI5 project.`

Reply:
Done.